src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -799,6 +800,14 @@ private bool canTailCall(CORINFO_METHOD_STRUCT_* callerHnd, CORINFO_METHOD_STRUC
             {
                 MethodDesc caller = HandleToObject(callerHnd);
 
+                if (caller.OwningType is EcmaType ecmaOwningType
+                    && ecmaOwningType.EcmaModule.EntryPoint == caller)
+                {
+                    // We don't want to tailcall the entrypoint for an application; It results in a rather
+                    // confusing debugging experience.
```

Comment:
```suggestion
                    // Do not tailcall from the application entrypoint.
                    // We want Main to be visible in stack traces.
```
We are disabling tailcalls *from* the entrypoint. Also, could you please update the same comment in the VM to match?